components/form/FAutocompleteInput.vue

```diff
@@ -0,0 +1,373 @@
+  focusColor?: Color;
+  /**
+   * Border focus color
+   */
+  focusBorderColor?: Color;
+  /**
+   * Background color of the options menu
+   */
+  optionsMenuColor?: Color;
+  /**
+   * Text color of option
+   */
+  optionTextColor?: Color;
+  /**
+   * Background color of the selected option
+   */
+  selectedOptionColor?: Color;
+  /**
+   * Text color of the selected option
+   */
+  selectedOptionTextColor?: Color;
+  /**
+   * Event that triggers validation
+   */
+  validationTrigger?: 'input' | 'change' | 'focus' | 'blur';
+  /**
+   * Loading state of the input
+   */
+  loading?: boolean;
+  /**
+   * Width of the options menu
+   */
+  menuWidth?: string | number;
+  /**
+   * Prevent item selection
+   */
+  preventSelection?: boolean;
+  /**
+   * Text displayed in menu when input is empty
+   */
+  emptyText?: string;
+  /**
+   * Text displayed in menu when no match for current input
+   */
+  noMatchText?: string;
+  /**
+   * Text displayed in menu when loading
+   */
+  loadingText?: string;
+  /**
+   * Can clear the current value
+   */
+  clearable?: boolean;
+}
+
+const props = withDefaults(defineProps<FAutocompleteInputProps>(), {
+  modelValue: '',
+  options: () => [],
+  label: '',
+  labelTextColor: 'neutral--dark-4',
+  name: '',
+  validateOnMount: false,
+  hint: '',
+  hintTextColor: 'neutral--dark-4',
+  hintIcon: null,
+  placeholder: '',
+  rules: () => [],
+  errorMessage: '',
+  preventSelection: false,
+  errorColor: 'danger',
+  color: 'neutral--light-3',
+  borderColor: 'secondary',
+  textColor: 'neutral--dark-4',
+  placeholderTextColor: 'neutral--dark-2',
+  optionsMenuColor: 'neutral--light-3',
+  optionTextColor: 'neutral--dark-3',
+  selectedOptionColor: 'primary--light-2',
+  selectedOptionTextColor: 'primary--dark-2',
+  validationTrigger: 'change',
+  menuWidth: 300,
+  outlineColor: 'neutral--light-3',
+  focusColor: 'neutral--light-5',
+  focusBorderColor: 'secondary',
+  emptyText: 'Type something to begin the search',
+  noMatchText: 'No match found for the current input.',
+  loadingText: 'Loading...',
+  clearable: false,
+});
+
+const emit = defineEmits<{
+  (name: 'update:modelValue', value: string | null): void;
+  (name: 'input', value: InputEvent): void;
+  (name: 'change', value: Event): void;
+  (name: 'focus', value: Event): void;
+  (name: 'blur', value: Event): void;
+  (name: 'new-value', value: string | null): void;
+}>();
+
+defineExpose<{
+  focus: () => void;
+}>({
+  focus,
+});
+
+const formattedFieldValue = computed({
+  get(): string {
+    if (!inputIsActive.value && !!currentOptionMatched.value) {
+      return currentOptionMatched.value.label;
+    }
+    return fieldValue.value;
+  },
+  set(inputValue: string) {
+    fieldValue.value = inputValue;
+    inputIsActive.value = true;
+    currentOptionMatched.value = undefined;
+  },
+});
+
+const inputIsActive = ref(false);
+
+const filterRegex = computed(
+  () => new RegExp(`(${escapeRegex(fieldValue.value)})`, 'ig')
```

Comment:
1) nice to use escapeRegex ;-)
2) we may need here to have a more powerfull regex by matching the diacritics (see compose-search-regex.js in Admin). We have a removeDiacritics in FDS I think BTW.
In Admin though, typing a name without diacritics won’t match one with diacritics, because they are in DB. However here, based on the items, you should match them by removing diacritions on all the labels too